text/0374-jsii-ts-version.md

```diff
@@ -0,0 +1,324 @@
+# The jsii compiler to follow TypeScript versioning
+
+* **Original Author(s)**: @RomainMuller
+* **Tracking Issue**: #374
+* **API Bar Raiser**: @{BAR_RAISER_USER}
+
+This RFC proposes to change the versioning scheme of the `jsii` compiler to stop
+conforming to [semantic versioning][semver], and instead use the `major.minor`
+version of the [TypeScript] compiler it is built on.
+
+## Working Backwards
+
+### Release Notes entry
+
+```md
+Starting with this release, the `jsii` compiler will no longer conform to
+[semantic versioning][semver]. Instead, its `major.minor` version will match
+that of the [TypeScript] compiler it uses (the [TypeScript] compiler does not
+conform to [semantic versioning][semver]).
```

Comment:
I don't think this is the most interesting thing about this decision. The most interesting thing is more directly framed as:

* jsii will now start tracking new TypeScript versions and will only release new features to the latest TypeScript version (not covered by the current wording)
* Users may have to change their code when they want to adopt new jsii versions (somewhat covered by implication by the current wording, but I think it's worthwhile to be explicit about it)

Reply:
We are looking at this, right?

![image](https://user-images.githubusercontent.com/524162/166455208-a57fd912-7096-46c4-9f0d-87838c725aea.png)

* If you want a newer jsii feature, you must take a newer TypeScript version
* If you want a newer TypeScript feature, you must take a newer jsii version

Reply:
Yeah that's correct.